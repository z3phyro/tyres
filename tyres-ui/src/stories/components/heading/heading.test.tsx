import { test, expect } from "vitest";
import { render } from "@solidjs/testing-library";
import Heading from "./heading";

test("Heading renders", () => {
  const { getByRole } = render(() => <Heading />);

  const heading = getByRole("heading") as HTMLElement;
  expect(heading).toBeTruthy();
});

test("Heading renders with custom content", () => {
  const { getByRole } = render(() => <Heading>Heading content</Heading>);

  const heading = getByRole("heading");
  expect(heading.textContent).toBe("Heading content");
});

test("Heading snapshot test", () => {
  const { asFragment } = render(() => <Heading>Heading content</Heading>);
  expect(asFragment()).toMatchSnapshot();
});


