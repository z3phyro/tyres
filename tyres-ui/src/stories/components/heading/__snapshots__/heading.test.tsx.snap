// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Heading snapshot test 1`] = `"<h1 class="text-2xl mb-4">Heading content</h1>"`;
